src/mailingui/components/badges/Badge.tsx

```diff
@@ -1,27 +1,31 @@
 import React, { CSSProperties, FC, ReactNode } from "react";
+import { useTheme } from "../../hooks/useTheme";
+import { BadgeVariantsKey } from "@mailingui/types";
 
 interface BadgeProps {
-  variant: "default" | "danger" | "success" | "info" | "warning";
-  size?: "sm" | "md" | "lg";
+  variant?: BadgeVariantsKey;
+  size?: keyof typeof sizes;
   dot?: boolean;
   pill?: boolean;
   noBorder?: boolean;
   children: ReactNode;
 }
 
 const Badge: FC<BadgeProps> = ({
-  variant,
+  variant = "default",
   size = "md",
   pill,
   dot,
   noBorder,
   children,
 }) => {
+  const theme = useTheme();
```

Comment:
Not a big fan of destructing everywhere but when using this `useTheme` hook should we make it a practice to destructing the theme to make our code a bit compact?